underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -125,29 +125,52 @@ public long skip(long n) throws IOException {
     if (n <= 0) {
       return 0;
     }
-    closeStream();
     mPos += n;
-    openStream();
     return n;
   }
 
   /**
-   * Opens a new stream at mPos if the wrapped stream mIn is null.
+   * Seek if we could reuse the existing stream or create a new one.
+   * @param length the content length to be read
    */
-  private void openStream() throws IOException {
-    if (mIn != null) { // stream is already open
+  private void seekStream(int length) throws IOException {
+    if (mIn != null && mPos == mInPos) { // stream is already open and at the expected position
       return;
     }
+    if (mIn != null && mPos > mInPos) { // stream is already open but the caller is seeking forward
+      long skip = mPos - mInPos;
+      if (skip < mIn.available()) {
+        try {
+          //TODO(beinan): we might need check the max skip length
+          if (skip == mIn.skip(skip)) {
```

Comment:
in the case skip returns a value which is not equal to `skip` (e.g. < skip), but not negative, the stream may still have more to available to skip. Should we try calling this in a loop? 

ref: https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#skip-long-

Reply:
the same question, we need to do something like the following.
```
do {
      skip(pos - mPos);
    } while (pos > mPos && in.available() > 0);
```

Reply:
@jia725 loop added